File: easy/differenceOfSums.py
class Solution:
    def differenceOfSums(self, n: int, m: int) -> int:
        return (n+1)*n//2-m*(n//m)*(n//m+1)
        
"""You are given positive integers n and m.

Define two integers as follows:

num1: The sum of all integers in the range [1, n] (both inclusive) that are not divisible by m.
num2: The sum of all integers in the range [1, n] (both inclusive) that are divisible by m.
Return the integer num1 - num2."""

File: easy/difference_of_sums.rs
//find differenc of the sums of integens less than n that are and are not divisible by m
impl Solution {
    pub fn difference_of_sums(n: i32, m: i32) -> i32 {
        n * (n + 1) / 2 - (n / m) * (n / m + 1) * m
    }
}
